src/sql/magic_cmd.py

```diff
@@ -51,6 +52,8 @@ def _validate_execute_inputs(self, line):
             "explore",
             "snippets",
         ]
+        CONNECTION_INDEPENDENT_COMMANDS = ["snippets"]
```

Comment:
a better name for this variable could be `COMMANDS_NO_CONNECTION_REQUIRED`, however, I see that in your `if` statements a few lines below you're using `not in`, this is harder to read than just `in`, try refactoring it to use `in`

e.g., for the connection required error, you could do `if cmd in COMMANDS_CONNECTION_REQUIRED and not ConnectionManager.current`